mayalookassigner/widgets.py

```diff
@@ -90,25 +93,76 @@ def get_selected_items(self):
     def get_all_assets(self):
         """Add all items from the current scene"""
 
-        self.clear()
-        items = commands.get_all_assets()
-        self.add_items(items)
+        with preserve_expanded_rows(self.view):
+            with preserve_selection(self.view):
+                self.clear()
+                nodes = commands.get_all_asset_nodes()
+                items = commands.create_items_from_nodes(nodes)
+                self.add_items(items)
 
         return len(items) > 0
 
     def get_selected_assets(self):
         """Add all selected items from the current scene"""
 
-        self.clear()
-        items = commands.get_items_from_selection()
-        self.add_items(items)
+        with preserve_expanded_rows(self.view):
+            with preserve_selection(self.view):
+                self.clear()
+                nodes = commands.get_selected_nodes()
+                items = commands.create_items_from_nodes(nodes)
+                self.add_items(items)
+
+    def get_nodes(self, selection=False):
+        """Find the nodes in the current scene per asset."""
+
+        items = self.get_selected_items()
+
+        # Collect all nodes by hash (optimization)
+        if not selection:
+            nodes = cmds.ls(dag=True,  long=True)
+        else:
+            nodes = commands.get_selected_nodes()
+        id_nodes = commands.create_asset_id_hash(nodes)
+
+        # Collect the asset item entries per asset
+        # and collect the namespaces we'd like to apply
+        assets = dict()
+        asset_namespaces = defaultdict(set)
+        for item in items:
+            asset = item["asset"]["name"]
+            asset_namespaces[asset].add(item.get("namespace"))
+
+            if asset in assets:
+                continue
+
+            assets[asset] = item
+            assets[asset]["nodes"] = id_nodes.get(str(item["asset"]["_id"]),
+                                                  [])
+
+        # Filter nodes to namespace (if only namespaces were selected)
+        for asset in assets:
+            namespaces = asset_namespaces[asset]
+
```

Comment:
Below this line we do NOTHING with `namespaces` if the value is `None`. In that case it would make more sense to do something like:

```
...
if None in namespaces:
     continue

nodes = assets[asset]["nodes"]
nodes = [node for node in nodes if
commands.get_namespace_from_node(node) in namespaces]
assets[asset]["nodes"] = nodes

return assets
```

Reply:
Haha, perfect @aardschok. Just beat you to it - just did that! :) fixed with https://github.com/Colorbleed/maya-look-assigner/pull/8/commits/ae66cebb5190ac13fb16a1a44b4880be458a4b7b